### PR TITLE
single particle filter, add needed features

### PR DESCRIPTION
**Pull request by alberto-sanchez**
I have added the possibility to check for antiparticles, making the input list of particles shorter, specially if I want to check for all b-hadrons. I also added some protection to the rapidity calculation, since in previous version was calculated for all particles in the event, even those with very small energy. In the process I add the option of debugging.

### CODE REVIEW

**Comment by perrozzi** [comment]
can you please add a _cfi and/or _cff in the python directory?

**Comment by perrozzi** [comment]
please test

**Comment by smuzaffar** [comment]
Looks like the commit is in future (tells me that it is 30 mins in future from now) that is why cms-bot is going crazy here.
I will check if I can fix cms-bot to avoid this kind of situation

**Comment by perrozzi** [comment]
thanks, I was going to ask during the ORP :-)

**Comment by smuzaffar** [comment]
please test
now https://github.com/cms-sw/cmssw/pull/15876/commits/059c16927e9788586654384fa98fa8a6ad862282 is in past, so cms-bot should start the test now

**Comment by alberto-sanchez** [comment]
@perrozzi , I have added the cfi for filtering major b-mesons.
@smuzaffar, sorry for all the mess caused with my clock in the future.

**Comment by perrozzi** [comment]
+1

File: GeneratorInterface/GenFilters/src/MCSingleParticleYPt.cc [hunk 1]
```diff
@@ -60,6 +62,16 @@ token_(consumes<edm::HepMCProduct>(edm::InputTag(iConfig.getUntrackedParameter("
        for (unsigned int i = 0; i < particleID.size(); i++){ defstat2.push_back(0);}
        status = defstat2;   
     } 
+
+    if (fVerbose > 0) {
+       std::cout <<     "----------------------------------------------------------------------" << std::endl;
```
Comment (davidlange6) [inline GeneratorInterface/GenFilters/src/MCSingleParticleYPt.cc]: @alberto-sanchez the costs should be replaced by LogWarning messages . thx.

Comment (alberto-sanchez) [inline GeneratorInterface/GenFilters/src/MCSingleParticleYPt.cc]: @davidlange6  have replace couts with LogWarning and LogInfo.

**Comment by alberto-sanchez** [comment]
@perrozzi, if nothing is wrong can you sign again?

**Comment by perrozzi** [comment]
+1

**Comment by perrozzi** [comment]
please test

**Comment by perrozzi** [comment]
@alberto-sanchez this still needs to be incorporated, right?

**Comment by alberto-sanchez** [comment]
@perrozzi  Yes, that is correct.

**Comment by perrozzi** [comment]
@davidlange6 could you please merge or comment?

**Pull request merged**